app/services/single_logout_handler.rb

```diff
@@ -52,6 +52,22 @@ def identity
     end
   end
 
+  def user
+    current_user || sp_slo_user
+  end
+
+  def sp_slo_user
+    sp_slo_identity&.user
+  end
+
+  def sp_slo_identity
+    @_sp_slo_identity ||= Identity.includes(:user).find_by(uuid: name_id)
+  end
+
+  def name_id
+    saml_request&.name_id
+  end
```

Comment:
great call! updated at f374f92

I think this might be a good thing to extract in the future as well. When working on this PR, the code in `SamlIdpController` `SamIdpLogoutConcern` and `SingleLogoutHandler` seemed to interact in many ways. Perhaps extracting some additional classes would make those easier to understand.